### PR TITLE
Recommend filenames for package examples

### LINKED ISSUE #413: Tell folks that examples show up on the pub site – if they are named correctly
See https://www.dartlang.org/tools/pub/package-layout#examples

The pub site will show example code for a package – see [here](https://pub.dartlang.org/packages/json_serializable#pub-pkg-tab-example).

But only if you name the example file correctly.

The current implementation is here: https://github.com/dart-lang/pub-dartlang-dart/blob/7431ab69e72283c11dbd9c03484b94d74e189a5d/app/lib/shared/utils.dart#L169

Includes:
`example/lib/main.dart`
`example/main.dart`
`example/lib/$package.dart`
`example/$package.dart`
`example/${package}_example.dart`
`example/example.dart`

### PR DESCRIPTION
**Pull request by kwalrath**
Staged: https://kw-www-dartlang-1.firebaseapp.com/tools/pub/package-layout#examples

Fixes #413

### CODE REVIEW

File: src/tools/pub/package-layout.md [hunk 1]
```diff
@@ -386,10 +386,27 @@ course! Those go inside the `example` directory. If the examples are complex
...
 
+If you might publish your package,
+consider creating an example file with one of the following names:
+
+* <code>example/lib/main.dart</code>
+* <code>example/main.dart</code>
+* <code>example/lib/<em>package_name</em>.dart</code>
+* <code>example/<em>package_name</em>.dart</code>
+* <code>example/<em>package_name</em>_example.dart</code>
+* <code>example/example.dart</code>
```
Comment (kevmoo) [inline src/tools/pub/package-layout.md]: also missed `lib/example.dart`

Comment (kwalrath) [reply src/tools/pub/package-layout.md]: Oops, and another one.

**Comment by kwalrath** [comment]
I'm a little leery of linking to the implementation, but I guess it's better than letting the docs possibly rot. However, just linking to where the files are listed isn't enough — you really need to look at where it's used. I put in the search I used, but maybe we could do something better?

**Comment by kwalrath** [comment]
@kevmoo PTAL

**Comment by chalin** [comment]
> I'm a little leery of linking to the implementation

FYI, the new code_excerpter can extract the `exampleFileCandidates` list of strings. (I haven't setup the new code_excerpter yet, but it is towards the top of my list of things to do for next week.)

**Comment by kwalrath** [comment]
I'm going to commit this. We can update the code/link later.

File: src/tools/pub/package-layout.md [hunk 2]
```diff
@@ -386,10 +386,31 @@ course! Those go inside the `example` directory. If the examples are complex
...
+* <code>example/<em>package_name</em>_example.dart</code>
+* <code>example/lib/example.dart</code>
+* <code>example/example.dart</code>
+
+When you publish a package that contains one of the above files,
+the pub site creates an **Example** tab to display that file.
+For example, the json_serializable package contains a file named
+`example/example.dart`, which appears in the
+[json_serializable Example tab.](https://pub.dartlang.org/packages/json_serializable#pub-pkg-tab-example)
+To see exactly how the example file is chosen, look at the
```
Comment (kevmoo) [inline src/tools/pub/package-layout.md]: Sorry for the late response. I just meant add a code comment here so any future maintainer of this page knows where the data comes from!

Then it isn't weird to point to a specific file at a specific commit, etc.

**Comment by kwalrath** [comment]
Ah, OK, much better. :) I'll make that fix.

**Pull request merged**